samples/maps/demo/data-class-two-ranges/demo.js

```diff
@@ -18,17 +18,43 @@
              * a pie chart showing the details for each state.
              */
             function pointClick() {
-                const row = this.options.row,
-                    $div = $('<div></div>')
-                        .dialog({
-                            title: this.name,
-                            width: 320,
-                            height: 300
-                        });
-
-                window.chart = new Highcharts.Chart({
+                const row = this.options.row;
+                const chart = this.series.chart;
+
+                // Remove existing annotation if present
+                chart.removeAnnotation('election-popup');
+
+                // Add new annotation with a pie chart
+                chart.addAnnotation({
+                    id: 'election-popup',
+                    labelOptions: {
+                        useHTML: true,
+                        backgroundColor: '#fff'
+                    },
+                    labels: [{
+                        point: this,
+                        text: `
+                            <div id="annotation-header">
+                                <span>${this.name}</span>
+                                <button id="annotation-close-btn">
+                                x
+                                </button>
+                            </div>
+                            <div id="popup-pie"></div>
+                        `,
+                        shape: 'rect'
+                    }],
+                    zIndex: 10
+                });
+
```

Comment:
I think we should set this annotation as a fixed popup, not dependent on the mouse position during clicking.

Reply:
I was considering that option, but then had a thought that maybe it's a better UX when the popup is not fixed but follows the clicked point; still when it's fixed it solves some positioning problems, so maybe it is the best UX after all :)